builder/main.py

```diff
@@ -142,6 +142,9 @@ def parse_args():
     parser.add_argument('--target', type=str, help="The target to cross-compile for (e.g. android-armv7, linux-x86, linux-aarch64)",
                         default='{}-{}'.format(current_os(), current_arch()),
                         choices=data.PLATFORMS.keys())
+    parser.add_argument('--cmake-extra', action='append', default=[],
```

Comment:
This could just be parsed in cmake.py, and that way you don't have to deal with changes here.

See: https://github.com/awslabs/aws-crt-builder/blob/main/builder/actions/install.py#L35 for an example

Reply:
Hmm.
I see your point modularity-wise
but that prevents the option from showing up in --help 😕
your call, but historically I've had a hard time knowing what the builder can or can't do

Reply:
We could give actions the ability to add to the global arg parser, or allow them to fill in help info when necessary. I agree, it's a half-measure, but I think allowing actions to have their own args is pretty important, especially for use-cases like aws-crt-python where the entire build process is in an injected python script.

Reply:
Moving it to cmake.py, adding to readme